### ISSUE TITLE
`dataclasses.make_dataclass` with a default field flagged as incompatible type

### ISSUE THREAD

**Issue by rsokl** [microsoft/pyright#1680]
**Describe the bug**
A standard use of `dataclasses.make_dataclass` is flagged when, e.g. `field(default=1)` is specified. The type checker states that `int` is incompatible with type `Field[Any]`

**To Reproduce**

```python
from dataclasses import make_dataclass
make_dataclass("A", fields=[("x", int, field(default=1))])
```

**Expected behavior**
I think that the above code should not be flagged by the type-checker

**Screenshots or Code**
![image](https://user-images.githubusercontent.com/29104956/112205731-774c5a00-8beb-11eb-9a7d-c67fc6fb86fa.png)

**VS Code extension or command-line**
Pylance: 2021.3.2
VSCode: 1.54.1

**Comment by erictraut** [comment]
I think pyright is doing the right thing here given the information in the typeshed `dataclasses.pyi` stub file. Mypy emits the same error for your code.

The problem appears to be in the typeshed definition for the `field` function. It returns a type that is incompatible with `Field[Any]`.

I recommend against using `make_dataclass` and instead use the class form, which is easier to read and avoids the problem you're seeing here.

```python
@dataclass
class A:
    x: int = field(default=1)
```

**Comment by rsokl** [comment]
Do you think that the typeshed folks would be receptive to fixing this if I opened an issue?

And I appreciate the recommendation, but I need to use `make_dataclass` to create dataclasses dynamically.

**Comment by erictraut** [comment]
I think you'll get quite a bit of pushback on this change in typeshed because all existing type checkers that support dataclasses rely on the current typeshed definition, so a "fix" in typeshed would precipitate breaking changes across all type checkers including mypy and pyright.
